Editor/EditorCore/ProBuilderAnalytics.cs

```diff
@@ -0,0 +1,183 @@
+// Will print event registration in the console as well as in editor.log
+//#define PB_ANALYTICS_LOGGING
+
+// Will allow data to be send if using a dev build
+//#define PB_ANALYTICS_ALLOW_DEVBUILD
+
+// Will allow data to be send from automated tests or scripts
+//#define PB_ANALYTICS_ALLOW_AUTOMATION
+
+// Data will stay locally and never sent to the database, the editor preference can also be used to prevent sending data
+//#define PB_ANALYTICS_DONTSEND
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+using UnityEditor;
+using UnityEditorInternal;
+using UnityEngine;
+using UnityEngine.Analytics;
+using UnityEngine.ProBuilder;
+
+namespace UnityEditor.ProBuilder
+{
+    static class ProBuilderAnalytics
+    {
+        static bool s_EventRegistered = false;
+        const int k_MaxEventsPerHour = 1000;
+        const int k_MaxNumberOfElements = 1000;
+        const string k_VendorKey = "unity.probuilder";
+        static string packageName = $"com.{k_VendorKey}";
+
+
+        // Data structure for Triggered Actions
+        [Serializable]
+        struct ProBuilderActionData
+        {
+            public string actionName;
+            public string actionType;
+            public string subLevel;
+            public int subLevelId;
+            public string triggeredFrom;
+        }
+
+        // Holds the type of data we want to send to the database
+        enum EventName
+        {
+            probuilderAction
+        }
+
+        // Triggered type is from where the action was performed
+        public enum TriggerType
+        {
+            MenuOrShortcut,
+            ProBuilderUI
+        }
+
+        // This will register all the Event type at once
+        private static bool RegisterEvents()
```

Comment:
style guide - omit redundant access modifiers. ex, `static bool RegisterEvents()` because `private` is implicit.